### PR TITLE
Add brave tracing categories

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/17900

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: patches/base-trace_event-builtin_categories.h.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/base/trace_event/builtin_categories.h b/base/trace_event/builtin_categories.h
+index 60bba6a85d393ddb19e954503c663b06244ad160..d268a1c848d6dcd5417e7341f16f8005bd448166 100644
+--- a/base/trace_event/builtin_categories.h
++++ b/base/trace_event/builtin_categories.h
+@@ -173,6 +173,7 @@
+   X("WebCore")                                                           \
+   X("webrtc")                                                            \
+   X("xr")                                                                \
++  BRAVE_INTERNAL_TRACE_LIST_BUILTIN_CATEGORIES(X) /*see choromium_src*/  \
```
Comment (iefremov) [inline patches/base-trace_event-builtin_categories.h.patch:9]: choromium -> chromium

File: chromium_src/base/trace_event/builtin_categories.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_BASE_TRACE_EVENT_BUILTIN_CATEGORIES_H_
+#define BRAVE_CHROMIUM_SRC_BASE_TRACE_EVENT_BUILTIN_CATEGORIES_H_
+
+// Please add all brave categories here. For small features use the existing
+// 'brave' category. For new big subsystems with a lot of traces create an
+// separated category named 'brave.<feature_name>`.
```
Comment (iefremov) [inline chromium_src/base/trace_event/builtin_categories.h:11]: I would add two more points:
 * pls read the original `builtin_categories.h` for more details
 * pls keep the list sorted

**Review by iefremov — APPROVED** [review]

File: patches/base-trace_event-builtin_categories.h.patch (hunk 1 above)
Comment (atuchin-m) [reply patches/base-trace_event-builtin_categories.h.patch]: done

File: chromium_src/base/trace_event/builtin_categories.h (hunk 2 above)
Comment (atuchin-m) [reply chromium_src/base/trace_event/builtin_categories.h]: added

File: chromium_src/base/trace_event/builtin_categories.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// named 'brave.<feature_name>`.
+// * Keep the list sorted.
+//
+// Note: THere is no #undef for this macros because it is used in exported
+// macros of the chromium builtin_categories.h.
+#define BRAVE_INTERNAL_TRACE_LIST_BUILTIN_CATEGORIES(X) \
+  X("brave")                                            \
+  X("brave.adblock")
+
+#include "../../../../base/trace_event/builtin_categories.h"
```
Comment (darkdh) [inline chromium_src/base/trace_event/builtin_categories.h:22]: shouldn't we use `src/base/trace_event/builtin_categories.h`

File: chromium_src/base/trace_event/DEPS [hunk 4]
```diff
@@ -0,0 +1,3 @@
+include_rules = [
+  "+../../../../base"
```
Comment (darkdh) [inline chromium_src/base/trace_event/DEPS:2]: this shouldn't be needed. cc @goodov

File: chromium_src/base/trace_event/builtin_categories.h (hunk 3 above)
Comment (goodov) [reply chromium_src/base/trace_event/builtin_categories.h:22]: yep, this PR was just created around the same time I merged my changes. @atuchin-m will fix it, no worries :)

Comment (atuchin-m) [reply chromium_src/base/trace_event/builtin_categories.h]: Done

File: chromium_src/base/trace_event/DEPS (hunk 4 above)
Comment (atuchin-m) [reply chromium_src/base/trace_event/DEPS]: removed.

File: chromium_src/base/trace_event/builtin_categories.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_BASE_TRACE_EVENT_BUILTIN_CATEGORIES_H_
+#define BRAVE_CHROMIUM_SRC_BASE_TRACE_EVENT_BUILTIN_CATEGORIES_H_
+
+// Please add all brave categories here. Take a look at the original
+// builtin_categories.h for more details. Please:
+// * For small features use the existing 'brave' category.
+// * For new big subsystems with a lot of traces create a separate category
+// named 'brave.<feature_name>`.
```
Comment (goodov) [inline chromium_src/base/trace_event/builtin_categories.h:13]: nit: wrong apostrophe at the end. should be `'brave.<feature_name>'`

**Review by goodov — APPROVED** [review]
++ with a small nit

File: chromium_src/base/trace_event/builtin_categories.h (hunk 5 above)
Comment (atuchin-m) [reply chromium_src/base/trace_event/builtin_categories.h:13]: `
```suggestion
// named 'brave.<feature_name>'.
```

**Pull request merged**